src/ttf.zig

```diff
@@ -0,0 +1,2904 @@
+        var substring: c.TTF_SubString = undefined;
+        try errors.wrapCallBool(c.TTF_GetTextSubStringForPoint(self.value, x, y, &substring));
+        return SubString.fromSdl(substring);
+    }
+
+    /// Get the previous substring in a text object
+    ///
+    /// ## Function Parameters
+    /// * `substring`: The `SubString` to query.
+    ///
+    /// ## Return Value
+    /// Returns the previous substring.
+    ///
+    /// ## Remarks
+    /// If called at the start of the text, this will return a zero length
+    /// substring with the `SubStringFlags.text_start` flag set.
+    ///
+    /// ## Thread Safety
+    /// This function should be called on the thread that created the text.
+    ///
+    /// ## Version
+    /// This function is available since SDL_ttf 3.0.0.
+    pub fn getPreviousSubString(self: Text, substring: SubString) !SubString {
+        var c_substring = substring.toSdl();
+        var previous: c.TTF_SubString = undefined;
+        try errors.wrapCallBool(c.TTF_GetPreviousTextSubString(self.value, &c_substring, &previous));
+        return SubString.fromSdl(previous);
+    }
+
+    /// Get the next substring in a text object
+    ///
+    /// ## Function Parameters
+    /// * `substring`: The `SubString` to query.
+    ///
+    /// ## Return Value
+    /// Returns the next substring.
+    ///
+    /// ## Remarks
+    /// If called at the end of the text, this will return a zero length substring
+    /// with the `SubStringFlags.text_end` flag set.
+    ///
+    /// ## Thread Safety
+    /// This function should be called on the thread that created the text.
+    ///
+    /// ## Version
+    /// This function is available since SDL_ttf 3.0.0.
+    pub fn getNextSubString(self: Text, substring: SubString) !SubString {
+        var c_substring = substring.toSdl();
+        var next: c.TTF_SubString = undefined;
+        try errors.wrapCallBool(c.TTF_GetNextTextSubString(self.value, &c_substring, &next));
+        return SubString.fromSdl(next);
+    }
+
+    /// Update the layout of a text object.
+    ///
+    /// ## Remarks
+    /// This is automatically done when the layout is requested or the text is
+    /// rendered, but you can call this if you need more control over the timing of
+    /// when the layout and text engine representation are updated.
+    ///
+    /// ## Thread Safety
+    /// This function should be called on the thread that created the text.
+    ///
+    /// ## Version
+    /// This function is available since SDL_ttf 3.0.0.
+    pub fn update(self: Text) !void {
+        return errors.wrapCallBool(c.TTF_UpdateText(self.value));
+    }
+};
+
+/// The representation of a substring within text.
+///
+/// ## Version
+/// This struct is available since SDL_ttf 3.0.0.
+pub const SubString = struct {
+    /// The flags for this substring
+    flags: SubStringFlags,
+    /// The byte offset from the beginning of the text
+    offset: c_int,
```

Comment:
Why `c_int`? There is no explicit cast, what is the intent of the SDL code, `usize`?